diff --git a/Makefile.am b/Makefile.am
index d18c49b8..f2ea648c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -914,7 +914,7 @@ SUFFIXES += .man
 	     -e "s|[@]MAN1EXT[@]|$(man1ext)|g" \
 	     -e "s|[@]MAN5EXT[@]|$(man5ext)|g" \
 	     -e "s|[@]MAN7EXT[@]|$(man7ext)|g" \
-	     -e "s|[@]MDATE[@]|`$(PERL) $(top_srcdir)/mdate.pl $<`|g" \
+	     -e "s|[@]MDATE[@]|`date +'%-d %B %Y' -r $(top_srcdir)/ChangeLog`|g" \
 	     -e "s|[@]OLDFONTDIR[@]|`echo $(oldfontdir) | sed -f $(makevarescape)`|g" \
 	     -e "s|[@]PDFDOCDIR[@]|`echo $(pdfdocdir) | sed -f $(makevarescape)`|g" \
 	     -e "s|[@]SYSTEMMACRODIR[@]|`echo $(systemtmacdir) | sed -f $(makevarescape)`|g" \
